### DISCUSSION TITLE
ERR_INVALID_URL - configured external url breaks startup process

### DISCUSSION THREAD

**Discussion by jdumeez** [immich-app/immich#13201] [Q&A]
### The bug

Since i installed the latest version of immich 1.116.2, the container no longer boots up.

Reproduction steps
Configure in server settings.
https://immich.app/docs/administration/system-settings#external-domain
Install latest version
Boot the add-on

(i run this as an add-on in home assistant - also raised defect here 
https://github.com/alexbelgium/hassio-addons/issues/1589 ) 

### The OS that Immich Server is running on

add-on 

### Version of Immich Server

1116.2

### Version of Immich Mobile App

1.0.0

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
https://github.com/alexbelgium/hassio-addons/tree/master/immich
```

### Your .env content

```Shell
https://github.com/alexbelgium/hassio-addons/tree/master/immich
```

### Reproduction steps

Reproduction steps
Configure in server settings.
https://immich.app/docs/administration/system-settings#external-domain

### Relevant log output

```shell
Addon Logs
api worker exited with code 1
Starting api worker
[Nest] 77057  - 09/28/2024, 7:16:38 PM     LOG [Api:EventRepository] Initialized websocket server
[Nest] 77057  - 09/28/2024, 7:16:38 PM   ERROR [Api:An instance of SystemConfigDto has failed the validation:
 - property server.externalDomain has failed the following constraints: isUrl 
] Validation error
TypeError: Invalid URL
    at new URL (node:internal/url:797:36)
    at SystemConfigCore.buildConfig (/app/immich/server/dist/cores/system-config.core.js:136:44)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)https://github.com/alexbelgium/hassio-addons/issues/new?assignees=&labels=bug&projects=&template=BUG-REPORT.yml&title=🐛+%5BAddon+name%5D+%3Ctitle%3E
    at async /app/immich/server/dist/cores/system-config.core.js:80:35 {
  code: 'ERR_INVALID_URL',
  input: 'immich.owndomain.xyz'
}
```

**Comment by bo0tzz** [comment]
You can downgrade to the previous version and fix it in the admin panel, or you can set a config file with a valid URL to get it to start up that way.

**Comment by jdumeez** [comment]
Hey boOtzz, thanks for the suggestion, but do note that the URL was working fine up until the update. (its a cloudflare tunnel) 
Can the config file be used to perform the specific URL change? Or does it overrule the entire config? 
I use an add-on of Immich within home assistant so cannot downgrade

**Comment by jdumeez** [comment]
@bo0tzz 
I started with part of the config file which did succesfully overrule the original url and i was able to boot up. 
Unfortunately, from the GUI now, i am not able to actually reset the original url.
Once i remove the config file, it is back to the same problem. 
So for now, I am now stuck with the config file option and can't change anything by gui anymore. 
Its seems the problem was that the URL didnt have https:// in front of the domain.  

<img width="955" alt="image" src="https://github.com/user-attachments/assets/49321a59-36e8-49f8-9c85-21a34b182f82">

**Comment by MahmoudAlyuDeen** [comment]
> or you can set a config file with a valid URL to get it to start up that way.

I ran into this error when moving from `config.json` -> to `config.yml` + `.env`

This worked in `config.json`:
```
"server": {
    "externalDomain": "https://my.domain.tld",
    "loginPageMessage": "Enter login information",
    "publicUsers": false
  },
```

But this fails with an `isUrl` validation under `config.yml`:

```
server:
  externalDomain: ${CONFIG_EXTERNAL_DOMAIN}
  loginPageMessage: Enter login information
  publicUsers: true
```

\+ .env: `CONFIG_EXTERNAL_DOMAIN=https://my.domain.tld`
\+ .env: `CONFIG_EXTERNAL_DOMAIN=https://my.domain.tld/`
& .env: `CONFIG_EXTERNAL_DOMAIN="https://my.domain.tld"`
& .env: `CONFIG_EXTERNAL_DOMAIN="https://my.domain.tld/"`

It seems .env substitution is not supported in config.yml:

<img width="814" alt="image" src="https://github.com/user-attachments/assets/1d22a929-5f4e-4c29-9edd-4b96f84c205e">
